python/tank/platform/bundle.py

```diff
@@ -823,9 +821,28 @@ def __resolve_setting_value(self, key, value):
             schema = self.__descriptor.get_configuration_schema().get(key)
         except:
             schema = None
-        
+
         if schema:
-            # post process against schema
-            value = self.__post_process_settings_r(key, value, schema)
+
+            # If the value is None, then the setting wasn't specified. Use the
+            # default value in the schema instead.
+            if value is None:
+
+                if schema.get("type") == "hook":
+                    # Hooks already have the concept of a default value. Just
+                    # use that special default hook string and allow the post
+                    # process to resolve the value correctly based on the
+                    # default value in the manifest. Because the hooks have a
+                    # bit of legacy path evaluation code, if we just use the
+                    # value from the manifest here, we aren't guaranteed to get
+                    # the expected path.
+                    value = constants.TANK_BUNDLE_DEFAULT_HOOK_SETTING
```

Comment:
From what I could tell, this setting triggers essentially the sparse config behavior (lookup the default value at runtime). If we just go with what's in the `info.yml` file, then the path evaluation doesn't work as (currently) expected. I think we need this for backward compatibility. I'll check with @manneohrstrom to verify my thought process.